src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/FindCommandParserTest.java

```diff
@@ -1,40 +1,266 @@
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_DUPLICATE_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_FROM_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_RANGE_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_SEARCH_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_DATE_TO_CONSTRAINTS;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.FindCommandParser.MESSAGE_EMPTY_TITLE_KEYPHRASE;
+import static ay2021s1_cs2103_w16_3.finesse.model.category.Category.MESSAGE_EMPTY_CATEGORY;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Predicate;
 
 import org.junit.jupiter.api.Test;
 
 import ay2021s1_cs2103_w16_3.finesse.logic.commands.FindCommand;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Date;
 import ay2021s1_cs2103_w16_3.finesse.model.transaction.Transaction;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.HasCategoriesPredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.HasExactAmountPredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.InAmountRangePredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.InDateRangePredicate;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.OnExactDatePredicate;
 import ay2021s1_cs2103_w16_3.finesse.model.transaction.predicates.TitleContainsKeyphrasesPredicate;
 
 public class FindCommandParserTest {
 
+    public static final String AMOUNT_FROM_DESC_BUBBLE_TEA = " " + PREFIX_AMOUNT_FROM + VALID_AMOUNT_BUBBLE_TEA;
+    public static final String AMOUNT_TO_DESC_BUBBLE_TEA = " " + PREFIX_AMOUNT_TO + VALID_AMOUNT_BUBBLE_TEA;
+    public static final String AMOUNT_FROM_DESC_INTERNSHIP = " " + PREFIX_AMOUNT_FROM + VALID_AMOUNT_INTERNSHIP;
+    public static final String AMOUNT_TO_DESC_INTERNSHIP = " " + PREFIX_AMOUNT_TO + VALID_AMOUNT_INTERNSHIP;
+    public static final String DATE_FROM_DESC_BUBBLE_TEA = " " + PREFIX_DATE_FROM + VALID_DATE_BUBBLE_TEA;
+    public static final String DATE_TO_DESC_BUBBLE_TEA = " " + PREFIX_DATE_TO + VALID_DATE_BUBBLE_TEA;
+    public static final String DATE_FROM_DESC_INTERNSHIP = " " + PREFIX_DATE_FROM + VALID_DATE_INTERNSHIP;
+    public static final String DATE_TO_DESC_INTERNSHIP = " " + PREFIX_DATE_TO + VALID_DATE_INTERNSHIP;
+
     private FindCommandParser parser = new FindCommandParser();
 
     @Test
-    public void parse_emptyArg_throwsParseException() {
-        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    public void parse_validTitleKeyphrases_success() {
```

Comment:
Possible to add tests for partial title keyphrases too?

Reply:
Will be covering this in `TitleContainsKeyphrasesPredicateTest`.